### PR TITLE
feat!: upgrade dependencies

### PR DESCRIPTION
**Pull request by ThisIsManta**
### Problems

`yarn audit` reported over 300+ vulnerabilities.

### How to verify the solution

Expect `pnpm` to be used instead of `yarn`
Expect `pnpm audit` to say "No known vulnerabilities found"
Expect `lib-es6` to replace the old `lib` directory. This mean moving forward, we target ES6 (ES2015) by default.
Expect this library to be more performant than [`reselect`](https://www.npmjs.com/package/reselect/v/5.1.1) according to `pnpm run bench`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -178,18 +165,13 @@ function createSelector(...funcs) {
...
+In “cache hit” scenarios, **rereselect** is faster than **Reselect**. The numbers below are in million-operation-per-second.
 
-Performance benchmarks are run as part of
-[CI build](https://circleci.com/gh/taskworld/rereselect). You can check out the
-latest results there.
+| Scenario | Reselect v3 | Reselect v5 | rereselect |
+|---|---|---|---|
+| cache hit (same state) | 15.6 | 16.6 | 17.1 |
+| cache hit (shallowly equal deps) | 5.5 | 1.1 | 7.6~9.8 |
+| cache miss | 3.9 | 0.5 | 3.8 |
```
Comment (ThisIsManta) [inline README.md:174]: Raw log from `pnpm run bench`

image

File: package.json [hunk 2]
```diff
@@ -37,5 +11,27 @@
...
+    "@types/lodash": "^4.17.6",
+    "lefthook": "^1.6.18",
+    "lodash": "^4.17.21",
+    "reselect3": "npm:reselect@^3.0.1",
+    "reselect5": "npm:reselect@^5.0.0",
+    "typescript": "^5.5.3",
+    "vitest": "^1.6.0"
+  },
+  "resolutions": {
+    "jsdom": "16.5.0"
```
Comment (ThisIsManta) [inline package.json:35]: Somehow `vitest` has `jsdom` v11 as its dependency which `jsdom` has many other vul.
Therefore I have to override `jsdom` here.

Note that `vitest` and `jsdom` are NOT part of the production code.

File: package.json [hunk 3]
```diff
@@ -1,34 +1,8 @@
...
-    "jest": "^23.5.0",
-    "lodash": "^4.17.10",
-    "matcha": "^0.7.0",
-    "prettier": "^1.14.2",
-    "reselect": "^3.0.1",
-    "ts-jest": "^23.1.4",
-    "typescript": "^3.0.1"
-  },
-  "name": "@taskworld.com/rereselect",
+  "name": "@taskworld/rereselect",
```
Comment (ThisIsManta) [inline package.json:2]: This is intentional. We have 2 NPM org: `taskworld` and `taskworld.com` which the first one is my perference.

image

File: package.json [hunk 4]
```diff
@@ -1,34 +1,8 @@
...
-    "prepare": "yarn clean && tsc && tsc --build tsconfig.es6.json",
-    "benchmark": "matcha benchmark.js"
-  },
-  "devDependencies": {
-    "@types/jest": "^23.3.1",
-    "@types/lodash": "^4.14.116",
-    "@types/node": "^10.7.1",
-    "jest": "^23.5.0",
-    "lodash": "^4.17.10",
-    "matcha": "^0.7.0",
```
Comment (ThisIsManta) [inline package.json:16]: `jest` and `matcha` (benchmark runner) are replaced by `vitest`, which `vitest` also offers `vitest bench` command.

File: package.json [hunk 5]
```diff
@@ -1,34 +1,8 @@
...
   "description": "A library that generates memoized selectors and supports dynamic dependency tracking.",
-  "version": "0.3.0",
-  "main": "lib/index.js",
-  "directories": {
-    "example": "examples",
-    "lib": "lib"
-  },
-  "dependencies": {},
-  "author": "Thai Pangsakulyanont <thai.p@taskworld.com> (http://dt.in.th/)",
+  "version": "1.0.0",
```
Comment (ThisIsManta) [inline package.json:4]: Once this PR is merged, I will publish the package onto NPM manually.
I'm too lazy to config a CI + NPM token. Plus, I don't expect we update this library often.

File: src/index.test.ts [hunk 6]
```diff
@@ -1,12 +1,13 @@
...
 import { memoize } from 'lodash'
+import { test, expect } from 'vitest'
 
 expect.addSnapshotSerializer({
   test(val) {
     return typeof val === 'function' && val.displayName
   },
   print(val) {
-    return `[Function ${val.displayName}]`
+    return `[Function ${(val as any).displayName}]`
```
Comment (y-nk) [inline src/index.test.ts:10]: a bit sad that vitest plugins are not implemented better in that regard

**Review by y-nk — APPROVED** [review]

**Comment by y-nk** [comment]
let's not publish just yet and use CI to do so

**Pull request merged**